sematic/types/types/tests/test_dataclass.py

```diff
@@ -129,9 +129,8 @@ def test_can_cast_type(from_type, to_type, expected_can_cast, expected_error):
             "Cannot cast field 'bad_dict_field' of "
             "BadDictField(bad_dict_field={'foo': 'bar'}) to "
             "<class 'sematic.types.types.tests.test_dataclass.BadDictField'>:"
-            " Dictionary doesn't have key/value types specified. Please use "
-            "'Dict[KType, VType]' instead of 'Dict' or 'dict'. "
-            "Dict[object, object] can be used for arbitrary dictionaries.",
+            " Type 'dict' is not a valid Sematic type: dict must be parametrized "
+            "(dict[...] instead of dict).",
```

Comment:
The lowercase `dict` spelling will be included in 3.8 and 3.9 executions as well, which will confuse users, as they would have to use `Dict` instead.

Can the actual resolved `typing` type (`dict` vs `Dict`) be resolved exactly, in order to construct a correct error message?

Reply:
the spelling that is used comes from the type that is used. If the users specified `foo(x: Dict)` the message would use `Dict`, but if they used `foo(x: dict)` it would use `dict`. The test here uses the latter.